src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs

```diff
@@ -0,0 +1,191 @@
+                                             ent-type (sel-ent "entityType")
+                                             disp (get-ordered-name sel-ent)]
+                                         (str "Permissions for " ent-type " " disp))]
+                               [:div {:style
+                                      {:float "left" :width column-width}}
+                                "User or Group ID"]
+                               [:div {:style
+                                      {:float "right" :width column-width}}
+                                "Access Level"]
+                               (common/clear-both)]
+                              (map-indexed
+                                (fn [i acl-entry]
+                                  [:div {}
+                                   (style/create-text-field
+                                     {:ref (str "acl-key" i)
+                                      :style {:float "left" :width column-width
+                                              :backgroundColor (when (< i (:count-orig @state))
+                                                                 (:background-gray style/colors))}
+                                      :disabled (< i (:count-orig @state))
+                                      :spellCheck false
+                                      :defaultValue (:userId acl-entry)})
+                                   (style/create-select
+                                     {:ref (str "acl-value" i)
+                                      :style {:float "right" :width column-width :height 33}
+                                      :defaultValue (:accessLevel acl-entry)}
+                                     access-levels)
+                                   (common/clear-both)])
+                                (:acl-vec @state))
+                              [comps/Button {:text "Add new" :style :add
+                                             :onClick #(do
+                                                        (react/call :capture-ui-state this)
+                                                        (swap! state update-in [:acl-vec] conj
+                                                          {:userId "" :accessLevel "READER"}))}]
+                              [:div {:style {:textAlign "center" :marginTop "1em"}}
+                               [:a {:href "javascript:;"
+                                    :style {:textDecoration "none"
+                                            :color (:button-blue style/colors)
+                                            :marginRight "1.5em"}
+                                    :onClick #((:dismiss-self props))}
+                                "Cancel"]
+                               [comps/Button {:text "Save"
+                                              :onClick #(do
+                                                         (react/call :capture-ui-state this)
```

Comment:
In the "capture-ui-state" the data is "captured" by loading it into the state (via the refs).... *but* it would seem possible to *instead* read the data from the UI(via the refs) and then simply return what is read (instead of loading it into the state).  What do you think?